src/textual/renderables/bar.py

```diff
@@ -112,13 +112,14 @@ def _apply_gradient(text: Text, gradient: Gradient, width: int) -> None:
     Args:
         text: A Text object.
         gradient: A Textual gradient.
-        width: Width to apply gradient to, or None for full width.
+        width: Width of gradient.
     """
     from_color = Style.from_color
     get_rich_color = gradient.get_rich_color
 
     max_width = width - 1
     if not max_width:
```

Comment:
Also just to check - if width == 0, then max_width will be -1 and we'll go into the loop which I don't think is intentional?

Reply:
The width should never be 0. It's the not the width of the renderable, but the range that is covered by the gradient.

Here it is the width of the renderable, and if that's 0, it won't render.

I could put in an assert, to validate that.

Reply:
But defensive code is good code. I'll add a handler for that case.

Reply:
Done.